org.civicrm.reqangular/src/common/components/date-picker.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/components',
+  'common/moment'
+], function (components, moment) {
+  components.component('datePicker', {
+    require: { ngModelCtrl: 'ngModel' },
+    bindings: {
+      clearBtnClass: '@',
+      showClearBtn: '<',
+      clearBtnClick: '&',
```

Comment:
why is `clearBtnClick` an external event and there are no internal events added to handle the clearing of the date picker button? 

On every implementation one should add the clear functionality? 

```html
<parent-cmp>
  <date-picker
    ...
    clear-btn-click="parentCtrl.clearDatePicker()"
  ></date-picker>    
</parent-cmp>
```

Reply:
Fixed this, if no event is passed, it clears the `ngModel`.